bindings/java/lib/src/main/java/org/iota/types/AccountHandle.java

```diff
@@ -556,6 +557,29 @@ public Transaction createAliasOutput(CreateAliasOutput options) throws WalletExc
         return CustomGson.get().fromJson(callAccountMethod(options), Transaction.class);
     }
 
+    /**
+     * Asks the faucet for funds.
+     *
+     * @throws NoFundsReceivedFromFaucetException when the faucet didn't fund the address.
+     */
+    public void requestFundsFromFaucet(RequestFundsFromFaucet options) throws WalletException, NoFundsReceivedFromFaucetException {
+        int maxAttempts = 5;
+        for(int i = 0; i < maxAttempts; i++) {
+            if(syncAccount(new SyncAccount()).getBaseCoin().getAvailable() < 1000000) {
```

Comment:
Who says that the user doesn't want to get more coins even if he has more than 1000000? :trollface:

Reply:
The faucet funds in 10 SMR right? Maybe better if we do like:

`if(syncAccount(new SyncAccount()).getBaseCoin().getAvailable() < 10000000)`

Reply:
My point is, let's say you have 1000000 already and you call this function, why should it return without requesting more funds from the faucet?
If you want to check with the amount it should be more like `int balanceBeforeFaucetCall = account.getBaseCoin().getAvailable()` and then later just check if the balance increased

Reply:
okay but the faucet won't fund in case there are already enough funds on the address.

> int balanceBeforeFaucetCall = account.getBaseCoin().getAvailable() and then later just check if the balance 

won't return a change then, and it would try 5 attempts (since of no change) and then throw an error.

Reply:
Then just return no error if you have balance
> The faucet funds in 10 SMR right?
I think that's what you get by default after the initial 1000, but if you run your own faucet, you can configure this value to something different, so it shouldn't be hardcoded I think

Reply:
> Then just return no error if you have balance
>  but if you run your own faucet, you can configure this value to something different, so it shouldn't be hardcoded I think

so to check for enough balance we need to do a check like `balance > targetAddressBalance`
do you think its fine with latest changes?